go/adbc/driver/internal/driverbase/connection.go

```diff
@@ -239,9 +260,16 @@ func (base *ConnectionImplBase) GetOptionInt(key string) (int64, error) {
 }
 
 func (base *ConnectionImplBase) SetOption(key string, val string) error {
-	switch key {
+	switch strings.ToLower(strings.TrimSpace(key)) {
```

Comment:
Same here - in general the caller shouldn't be adding spaces...

Reply:
Removed unnecessary `TrimSpace` calls